### DISCUSSION TITLE
Encryption/Decryption of a specific column

### DISCUSSION THREAD

**Discussion by shubhendumadhukar** [pocketbase/pocketbase#2155] [Q&A]
I am trying to encrypt and decrypt the contents of a specific column using the event hooks. For example, in `BeforeRecordCreateEvent`, I use my encrypt function to modify the text like
```go
encrypted_text, err := utils.Encrypt([]byte(e.Record.GetString("message")))
if err != nil {
  return err
}
e.Record.Set("message", string(encrypted_text[:]))
```
This works fine. And I was hoping to decrypt the column in `OnRecordsListRequest` as:
```go
for _, record := range e.Records {
  decrypted, err := utils.Decrypt([]byte(record.GetString("message")))
  if err != nil {
	fmt.Println(err)
  }
  fmt.Println(string(decrypted[:]))
  record.Set("message", string(decrypted[:]))
}
```
I assumed that this will send the decrypted text to my frontend and not change anything in the database (because I am not calling anything like `app.Dao().SaveRecord`). However, `record.Set("message", string(decrypted[:]))` seems to modify the records in the DB too.

Is there a way I can use the hooks to only modify the response going to the client, rather than modifying the records in the DB too?

**Comment by ganigeorgiev** [comment]
Changing the `e.Record` props in the `OnRecordsListRequest` and `OnRecordViewRequest` hooks doesn't modify the record in the database.

`record.Set()` just replaces the field value with the user provided one. If you want to persist the change you'll need to call `app.Dao().SaveRecord()` (this is not necessarily only for the create and update hooks since the finalizer will call `SaveRecord()` anyway.)

If that's not what you are experiencing, please provide a more detailed code sample.

**Reply by shubhendumadhukar** [reply]
Okay, I think the misunderstanding is because of what I see in the admin UI. I queried the DB, and it seems the DB has the encrypted values. But since admin UI is probably calling the same APIs, `OnRecordsListRequest` is triggered there too, leading to my assumption that the values in the DB have changed. Probably not a bug, but a bit confusing.

**Reply by ganigeorgiev** [reply]
Yes, the Admin UI also uses the same list and view APIs.

If you want to exclude the admins, you can do the decryption only for authenticated regular app users and leave the default behavior for everything else.

Or alternatively, depending on your use case, you can also perform the decryption only if there is some special query parameter send with the request.
